drivers/sensor/ti/tmp11x/tmp11x.c

```diff
@@ -91,8 +87,24 @@ static bool check_eeprom_bounds(const struct device *dev, off_t offset,
 	return true;
 }
 
-int tmp11x_eeprom_write(const struct device *dev, off_t offset,
-			const void *data, size_t len)
+int tmp11x_eeprom_await(const struct device *dev)
+{
+	int res;
+	uint16_t val;
+
+	k_sleep(K_MSEC(EEPROM_MIN_BUSY_MS));
+
+	do {
```

Comment:
using the WAIT_FOR macro with a time limit on trying this here would be quite sensible